longhorn/longhorn-manager
# Require file check if the bids in failed state

**Problem:**
backingimage and backingimagedatasource deletion got stuck, if the backing image is deleted during upload process and bids just changes to `ready-for-transfer` state

**Solution:**
For bim controller, it requires file check as long as the bids in failed state

**Related Issue:**
[https://github.com/longhorn/longhorn/issues/6086](https://github.com/longhorn/longhorn/issues/6086)

**Test Plan:**
- Via Longhorn GUI, upload backing image from local ![bi_upload](https://github.com/longhorn/longhorn/assets/19387129/7309a630-78b3-4f4c-b24c-1962220cadf5)
- Delete backingimage as bids just turns into `ready-for-transfer` state 
  - Delete backingimage by `$ kubectl -n longhorn-system delete backingimage img1 --cascade=foreground` ![bi_delete_ready_for_transfer](https://github.com/longhorn/longhorn/assets/19387129/e7fb5e7f-e2a0-4178-adc0-253f94d36f56)
- CR backingimage, backingimagedatasource and backingimagemanager should be cleared
